src/classes/BDI_MappingServiceAdvanced.cls

```diff
@@ -278,8 +305,87 @@ public with sharing class BDI_MappingServiceAdvanced implements BDI_MappingServi
+        Map<String,BDI_FieldMapping> fieldMappingByDevName,
+        Map<String,BDI_FieldMapping[]> fieldMappingsByObjMappingDevName) {
+
+
+        // TODO: Turn magic strings below into constants once custom labels/names are decided.
+        BDI_ObjectMapping simulatedObjectMapping =
+            constructSimulatedObjectMapping('Donor_Fields', 'Donor Fields', SObjectType.DataImport__c.Name);
+        BDI_FieldMapping donationDonorField =
+            constructSimulatedFieldMapping(
+                SObjectType.DataImport__c.fields.Donation_Donor__c,
+                simulatedObjectMapping);
+
+        BDI_FieldMapping[] simulatedFieldMappings = new BDI_FieldMapping[] {donationDonorField};
+
+        objectMappingByDevName.put(simulatedObjectMapping.DeveloperName, simulatedObjectMapping);
+        fieldMappingByDevName.put(donationDonorField.DeveloperName, donationDonorField);
+        fieldMappingsByObjMappingDevName.put(simulatedObjectMapping.DeveloperName, simulatedFieldMappings);
+    }
+
+    /*******************************************************************************************************
+    * @description Method constructs an instance of BDI_ObjectMapping with the bare minimum required
+    * fields for use in the addUnmappedFields method. This is to support adding in fields that aren't
+    * Field Mappings, but need to look like Field Mappings such as the DataImport__c.Donation_Donor__c field.
+    *
+    * @param developerName: Object Mapping developer name.
+    * @param masterLabel: Object Mapping master label, this will display in the UI.
+    * @param objectApiName: The target sObject API name.
+    */
+    private static BDI_ObjectMapping constructSimulatedObjectMapping(String developerName,
+        String masterLabel,
+        String objectApiName) {
+
+        BDI_ObjectMapping objectMapping = new BDI_ObjectMapping();
+        objectMapping.DeveloperName     = developerName;
+        objectMapping.MasterLabel       = masterLabel;
+        objectMapping.Object_API_Name   = objectApiName;
+
+        return objectMapping;
+    }
+
+    /*******************************************************************************************************
+    * @description Method constructs an instance of BDI_FieldMapping with the bare minimum required
+    * fields for use in the addUnmappedFields method. This is to support adding in fields that aren't
+    * Field Mappings, but need to look like Field Mappings such as the DataImport__c.Donation_Donor__c field.
+    *
+    * @param fieldDescribe: Describe info of the field we need a Field Mapping for.
+    * @param parentObjectMapping: The parent object mapping this Field Mapping needs to reference.
+    */
+    private static BDI_FieldMapping constructSimulatedFieldMapping(Schema.DescribeFieldResult fieldDescribe,
+        BDI_ObjectMapping parentObjectMapping) {
+
+        BDI_FieldMapping fieldMapping   = new BDI_FieldMapping();
```

Comment:
not like the BIGGEST fan of aligning the equal signs like this but might bet willing to let it slide - not sure if we have guidance around this in our style guide or not